Gordon360/ApiControllers/ProfilesController.cs

```diff
@@ -361,33 +361,52 @@ public IHttpActionResult GetCliftonStrengths(string username)
         public IHttpActionResult GetMyImg()
         {
             var authenticatedUser = this.ActionContext.RequestContext.Principal as ClaimsPrincipal;
+            var username = authenticatedUser.Claims.FirstOrDefault(x => x.Type == "user_name").Value;
             var id = authenticatedUser.Claims.FirstOrDefault(x => x.Type == "id").Value;
-            var viewerName = authenticatedUser.Claims.FirstOrDefault(x => x.Type == "user_name").Value;
-            var viewerType = _roleCheckingService.getCollegeRole(viewerName);
-            var photoInfo = _profileService.GetPhotoPath(id);
+
+            var photoModel = _profileService.GetPhotoPath(id);
+
             string pref_img = "";
             string default_img = "";
+           
             var fileName = "";
             var filePath = System.Web.Configuration.WebConfigurationManager.AppSettings["DEFAULT_PREF_IMAGE_PATH"];
+
+            var unapprovedFileName = username + "_" + _accountService.GetAccountByUsername(username).account_id + ".jpg";
+            var unapprovedFilePath = System.Web.Configuration.WebConfigurationManager.AppSettings["DEFAULT_ID_SUBMISSION_PATH"];
+
             byte[] imageBytes;
             JObject result = new JObject();
 
-            //return default image if no photo info found for this user.
-            if (photoInfo == null)
+            if (photoModel != null)
+            {
+                // If this is the case, we just set the filename and complete the process after this if-else block
+                fileName = photoModel.Pref_Img_Name;
+            }
+            else if (File.Exists(unapprovedFilePath + unapprovedFileName))
+            {
+
+                var webClient = new WebClient();
+                imageBytes = webClient.DownloadData(unapprovedFilePath + unapprovedFileName);
+
+                string unapproved_img = Convert.ToBase64String(imageBytes);
+                result.Add("unap", unapproved_img);
+                return Ok(result);
+            }
+            else
             {
                 var webClient = new WebClient();
                 imageBytes = webClient.DownloadData(System.Web.Configuration.WebConfigurationManager.AppSettings["DEFAULT_PROFILE_IMAGE_PATH"]);
                 default_img = Convert.ToBase64String(imageBytes);
                 result.Add("def", default_img);
                 return Ok(result);
             }
-            else
-                fileName = photoInfo.Pref_Img_Name;
+                
```

Comment:
It's unintuitive to me that the first case of the `if` block is the expected behavior that continues after the block, whereas the other two return early from within the block. I think it would be clearer if this `if` block explicitly handles the two edge cases that return early, and the 'normal' path continues afterward:
```
if (photoModel == null)
            {
                if (File.Exists(unapprovedFilePath + unapprovedFileName))
                {

                    var webClient = new WebClient();
                    imageBytes = webClient.DownloadData(unapprovedFilePath + unapprovedFileName);

                    string unapproved_img = Convert.ToBase64String(imageBytes);
                    result.Add("unap", unapproved_img);
                    return Ok(result);
                }
                else
                {
                    var webClient = new WebClient();
                    imageBytes = webClient.DownloadData(System.Web.Configuration.WebConfigurationManager.AppSettings["DEFAULT_PROFILE_IMAGE_PATH"]);
                    default_img = Convert.ToBase64String(imageBytes);
                    result.Add("def", default_img);
                    return Ok(result);
                }
            }

            fileName = photoModel.Pref_Img_Name;
```